Address Blair's comments from our zoom meeting including adding a discussion paragraph about future graph algorithms work (#8)
taylorreiter/2021-paper-ibd: content/04.discussion.md

+ assembly graphs can be built with a combination of long and short reads or from long or short reads alone, so dominating set differential abundance analysis is a flexible method to compare groups of metagenomes that will accommodate the field as metagenome sequencing continues to improve
+ the framework is theoretically extensible to metatranscriptomes, but this remains a point of future research
-->

While we present an initial pipeline that enables differential abundance analysis directly on assembly graphs, we identified several areas where our approach could be improved.
First, implementing approaches to better control graph piece size would be beneficial.

> We heuristically selected *r* = 10 to build the metapangeome graphs because this radius size produced dominating set pieces with approximately the same number of k-mers as the average bacterial gene for pieces that were present in many metagenomes.

Reviewer: ```suggestion
We heuristically selected *r* = 10 to build the metapangenome graphs because this radius produced dominating set pieces with approximately the same number of k-mers as the average bacterial gene for pieces that were present in many metagenomes.
```

> However, the radius size needed to meet this condition may change depending on the diversity of the sequencing reads used to build the metapangenome graph.

Reviewer: ```suggestion
However, the radius needed to meet this condition may change depending on the diversity of the sequencing reads used to build the metapangenome graph.
```

Author: I like your suggestion! I'll change throughout, thank you!

Diversity increases with the complexity of the sequenced community, sequencing depth, and the number of communities observed.
Algorithms that either automatically select a radius size that achieves a user-specified average piece size, or that produce more consistent piece sizes independent of diversity of the sequencing data would provide finer control of the graph structure and subsequent sequence comparisons.

> Switching from a cDBG to a de Bruijn graph as the base spacegraphcats graph structure could lead to more consistent piece sizes; cDBGs have variable node sizes because they combine nodes without branching paths, while every node in a de Bruijn graph is length *k*. 

Reviewer: Maybe "Further, switching..." ?

Author: ```suggestion
Further, switching from a cDBG to a de Bruijn graph as the base spacegraphcats graph structure could lead to more consistent piece sizes; cDBGs have variable node sizes because they combine nodes without branching paths, while every node in a de Bruijn graph is length *k*. 
```

> Switching from a cDBG to a de Bruijn graph as the base spacegraphcats graph structure could lead to more consistent piece sizes; cDBGs have variable node sizes because they combine nodes without branching paths, while every node in a de Bruijn graph is length *k*. 

Reviewer: It seems weird to talk about "length" for a node in this context - at least, I think of them as being associated with sets of k-mers, not subsequences of variable length (though I realize these are interchangeable given the way that things are consolidated).

Second, improving RAM efficiency at high radii would enable more diversity to be represented in individual graphs.
In order to build the metapangenome graphs, we first hard-trimmed the input sequences to remove low abundance k-mers, thereby decreasing the RAM needed to construct each graph.
Algorithmic changes that improve RAM efficiency at high radii would obviate the need for hard trimming, and increase the amount of diversity that could be represented in a single graph. 

> Similarly, it may allow dominating set differential abundance analysis to be performed directly on groups of metagenomes without the need to first identify species of interest via genome queries.

Reviewer 2: ```suggestion
Similarly, improved performance would allow dominating set differential abundance analysis to be performed directly on groups of metagenomes without the need to first identify species of interest via genome queries.
```
